src/WebJobs.Extensions.DurableTask/LocalHttpListener.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.WebApiCompatShim;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    internal class LocalHttpListener : IDisposable
+    {
+        private readonly DurableTaskExtension extension;
+        private readonly IWebHost localWebHost;
+        private readonly Func<HttpRequestMessage, Task<HttpResponseMessage>> handler;
+
+        public LocalHttpListener(
+            DurableTaskExtension extension,
+            Uri listenUri,
+            Func<HttpRequestMessage, Task<HttpResponseMessage>> handler)
+        {
+            this.extension = extension ?? throw new ArgumentNullException(nameof(extension));
+            this.handler = handler ?? throw new ArgumentNullException(nameof(handler));
+            this.localWebHost = this.CreateWebHost(listenUri);
+        }
+
+        public bool IsListening { get; private set; }
+
+        public void Dispose()
+        {
+            this.localWebHost?.Dispose();
```

Comment:
Because we will initialize `localWebHost` to `null` if this is running in Functions v1.

One could argue that we shouldn't even go down this code path at all for Functions v1, but my concern is that would make the end-to-end code even harder to reason about due to the additional conditional compilation.

Reply:
What if in `this.CreateWebHost()` we just create a dummy `IWebHost` that no-ops on all of it's interface methods? It would make the only place we need conditional compilation that method, and reduce the likelihood of null reference exceptions.

Reply:
That's a good suggestion. Let me look into that.